### PR TITLE
Utilizando novo endpoint dos Correios

### LINKED ISSUE #4: Correios fora do ar
Alguém sabe dizer se nessa semana os Correios estão fora do ar para usar rastreamento ou se mudou algo que deverá ser ajustado?

### PR DESCRIPTION
**Pull request by lemanchester**
Fix #4 

 - Tentei manter a compatibilidade com a última versão da gem.

Aguardo seus comentários @prodis

### CODE REVIEW

File: lib/correios/sro/object.rb [hunk 1]
```diff
@@ -6,7 +6,14 @@ class Object
       include SAXMachine
 
       element  :numero, :as => :number
+      element  :erro,   :as => :error
       elements :evento, :as => :events, :class => Correios::SRO::Event
+
+
```
Comment (prodis) [inline lib/correios/sro/object.rb]: Blank line.

File: lib/correios/sro/parser.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
+require "pry"
```
Comment (prodis) [inline lib/correios/sro/parser.rb]: Pry.

File: lib/correios/sro/xml_builder.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+module Correios
+  module SRO
+    class XMLBuilder
+
+      QUERY_TYPES =  { list: "L", range: "F" }
+
+      RESULT_MODES = { all:  "T", last:  "U" }
+
+      LANGUAGE = { pt:  "101", en:  "102" }
```
Comment (prodis) [inline lib/correios/sro/xml_builder.rb]: Please freeze the constants.

File: lib/correios/sro/xml_builder.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      end
+
+      private
+
+      def namespaces
+        {
+          "xmlns:soapenv" => "http://schemas.xmlsoap.org/soap/envelope/",
+          "xmlns:res" => "http://resource.webservice.correios.com.br/"
+        }
+      end
```
Comment (prodis) [inline lib/correios/sro/xml_builder.rb]: Should be namespaces constants?

File: lib/correios/sro/xml_builder.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
+module Correios
+  module SRO
+    class XMLBuilder
```
Comment (prodis) [inline lib/correios/sro/xml_builder.rb]: How about a better name that reflects the responsibility of this class? Maybe changing `build` method for something like `build_request` or `build_request_body` is enough. 

What do you think?

**Comment by prodis** [comment]
Please see my comments.

Thank you.

**Comment by lemanchester** [comment]
thanks @prodis I agree with the comments, let me know what if that reflects your thoughts.

**Comment by prodis** [comment]
@lemanchester Great! 👍 

Please squash all changes into one commit. After I will merge it.

**Comment by lemanchester** [comment]
Hey @prodis just want to make sure, do you want that I squash all commits into one?

**Comment by prodis** [comment]
@lemanchester Yes.

**Comment by lemanchester** [comment]
hey @prodis let me know if that works. 

thx

**Comment by angelim** [comment]
I think that travis-ci error can be fixed having it install a newer version of Bundler. There's a [question](http://stackoverflow.com/questions/36979283/how-to-fix-travis-error-nomethoderror-undefined-method-spec-for-nilnilclass) on SO about this.

Adding the following to .travis.yml might fix it:
```yml
before_install:
  - gem install bundler
```

**Comment by angelim** [comment]
`coveralls`, which is a development dependency, needs `json >= 1.8` and bundler tries to get the latest version. json 2 doesn't support ruby 1.9.3.
I think this is up to @prodis but you can either manually add json 1.8 to the gem's development dependencies or drop support for 1.9.3 altogether.
Specs run fine with json 1.8

**Comment by prodis** [comment]
@lemanchester 
Really sorry for delay.
"Many things, all together, at the same time, right now" happening in my life.

**Comment by angelim** [comment]
Thanks @prodis and @lemanchester

**Pull request merged**